src/Nethermind/Nethermind.Monitoring/MonitoringService.cs

```diff
@@ -80,7 +80,8 @@ public async Task StartAsync()
             }
             if (_exposePort is not null)
             {
-                IMetricServer metricServer = new KestrelMetricServer(_exposePort.Value);
+                // KestrelMetricServer intercept SIGTERM causing exitcode to be incorrect
+                IMetricServer metricServer = new MetricServer(_exposePort.Value);
```

Comment:
Are there any implications because of this change?

Reply:
I'm guessing its a bit spower compared to kestral. Metrics seems to still works.

Reply:
It disables non system grafana metrics for Windows 😢

As it uses HttpListener and you need to explicitly give it permissions (outside in OS) to listen on that port or run in Admin mode.

Perhaps switch on OS as I assume Windows isn't paying attention to `SIGTERM`?

Reply:
https://github.com/NethermindEth/nethermind/pull/6326